karma.conf.js

```diff
@@ -53,63 +64,77 @@ var customLaunchers = {
   }
 };
 
-var options = {
+var baseOptions = {
 
-  browserDisconnectTimeout: 10000,
-  browserDisconnectTolerance: 2,
-  browserNoActivityTimeout: 90000,
-  captureTimeout: 120000,
+  browserify: {
+    configure: function(bundle) {
+      bundle.on('prebundle', function() {
+        dependencies.forEach(function(name) {
+          bundle.require(depMain(name), {expose: name});
+        });
+      });
+    }
+  },
 
   client: {
     mocha: {opts: 'test/mocha.opts'}
   },
 
-  plugins: [
-    require('karma-browserify'),
-    require('karma-mocha'),
-    require('karma-sauce-launcher')
-  ],
+  files: dependencies.map(depMain).concat(['index.js', 'test/**/*.js']),
 
   frameworks: [
     'browserify',
     'mocha'
   ],
 
-  files: dependencies.map(depMain).concat(['index.js', 'test/**/*.js']),
+  plugins: [
+    require('karma-sauce-launcher'),
+    require('karma-browserify'),
+    require('karma-mocha')
+  ],
 
   preprocessors: {
     'test/**/*.js': ['browserify']
   },
 
+  singleRun: true
+
+};
+
+var remoteOptions = Object.assign({
+
+  browsers: Object.keys(customLaunchers),
+
+  browserDisconnectTimeout: 10000,
+  browserDisconnectTolerance: 2,
+  browserNoActivityTimeout: 90000,
+  captureTimeout: 120000,
+
+  customLaunchers: customLaunchers,
+
+  plugins: [
+    require('karma-browserify'),
+    require('karma-mocha'),
+    require('karma-sauce-launcher')
+  ],
+
   reporters: [
     'dots',
     'saucelabs'
   ],
 
-  singleRun: true,
-
-  browserify: {
-    configure: function(bundle) {
-      bundle.on('prebundle', function() {
-        dependencies.forEach(function(name) {
-          bundle.require(depMain(name), {expose: name});
-        });
-      });
-    }
-  },
-
   sauceLabs: {
     testName: 'Sanctuary',
     project: 'Sanctuary',
     name: 'Sanctuary Test Suite',
     startTunnel: true,
     timeout: 600
-  },
-
-  customLaunchers: customLaunchers,
+  }
 
-  browsers: Object.keys(customLaunchers)
+}, baseOptions);
 
+module.exports = function(config) {
+  process.env.CI === 'true' ?
+    config.set(remoteOptions) :
+    config.set(baseOptions);
```

Comment:
Slightly cleaner:

```javascript
config.set(process.env.CI === 'true' ? remoteOptions : baseOptions);
```

Also, we could add [envvar][1] as a dev dependency:

```javascript
var CI = envvar.boolean('CI', false);
```


[1]: https://github.com/plaid/envvar